packages/esm-patient-common-lib/src/workspaces/workspaces.ts

```diff
@@ -103,40 +103,28 @@ function getTitleFromExtension(ext: ExtensionRegistration) {
   return ext.name;
 }
 
+function getWhetherWorkspaceCanBeClosed(name: string, ignoreChanges: boolean = false) {
+  const promptBeforeFn = getPromptBeforeClosingFcn(name);
+  return ignoreChanges || !promptBeforeFn || !promptBeforeFn();
```

Comment:
Less clever, but more readable, I think (plus this skips the function call if `ignoreChanges` is `true`).

```suggestion
function getWhetherWorkspaceCanBeClosed(name: string, ignoreChanges: boolean = false) {
  if (ignoreChanges) {
    return true;
  }
  
  const promptBeforeFn = getPromptBeforeClosingFcn(name);
  return !promptBeforeFn || !promptBeforeFn();
```